src/GafferUI/StandardConnectionGadget.cpp

```diff
@@ -188,24 +188,46 @@ void StandardConnectionGadget::doRenderLayer( Layer layer, const Style *style )
 
 	const_cast<StandardConnectionGadget *>( this )->setPositionsFromNodules();
 
+	const Gadget *srcNodeGadget;
+	if( !srcNodule() )
+	{
+		const Plug *srcPlug = dstNodule()->plug()->getInput();
+		const GraphGadget *graphGadget = parent<GafferUI::GraphGadget>();
+		srcNodeGadget = graphGadget->nodeGadget( srcPlug->node() );
+	}
+
 	Style::State state = ( m_hovering || m_dragEnd || m_dotPreview ) ? Style::HighlightedState : Style::NormalState;
-	if( state != Style::HighlightedState && srcNodule() && dstNodule() )
+
+	if( state != Style::HighlightedState)
 	{
-		const Gadget *srcNodeGadget = srcNodule()->ancestor<NodeGadget>();
+		if( srcNodule() )
+		{
+			srcNodeGadget = srcNodule()->ancestor<NodeGadget>();
+		}
 		if( srcNodeGadget && srcNodeGadget->getHighlighted() )
 		{
 			state = Style::HighlightedState;
 		}
-		else
+	}
+
+	if( state != Style::HighlightedState && dstNodule())
+	{
+		const Gadget *dstNodeGadget = dstNodule()->ancestor<NodeGadget>();
+		if( dstNodeGadget && dstNodeGadget->getHighlighted() )
 		{
-			const Gadget *dstNodeGadget = dstNodule()->ancestor<NodeGadget>();
-			if( dstNodeGadget && dstNodeGadget->getHighlighted() )
-			{
-				state = Style::HighlightedState;
-			}
+			state = Style::HighlightedState;
 		}
 	}
 
+	// If we're missing the srcNodule, the source node is either hidden or we're
+	// dealing with an AuxiliaryConnection.
+	if( !srcNodule() && srcNodeGadget )
+	{
+		Box3f b = srcNodeGadget->transformedBound();
+		style->renderAuxiliaryConnection( V2f( b.center().x, b.center().y ), V2f( m_srcTangent.x, m_srcTangent.y ), V2f( m_dstPos.x, m_dstPos.y ), V2f( m_dstTangent.x, m_dstTangent.y ), state );
+		return;
```

Comment:
This early return skips the drawing of the extra connection when `m_addingConnection == true`. Actually, I'm a bit confused, but I think your branch is from a point before we merged the `m_addingConnection` stuff?